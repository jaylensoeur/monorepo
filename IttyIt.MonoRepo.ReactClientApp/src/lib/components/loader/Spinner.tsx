import CircularProgress from "@material-ui/core/CircularProgress";
import * as React from "react";

type Props = {
    isLoading: boolean
    className?: string
}

const Spinner = ({isLoading, className}: Props): JSX.Element => {

    if (isLoading) {
        return (
            <div data-testid="spinner" className={className}>
                <CircularProgress color='secondary'/>
            </div>
        )
    }

    return null;
};

export default Spinner;
