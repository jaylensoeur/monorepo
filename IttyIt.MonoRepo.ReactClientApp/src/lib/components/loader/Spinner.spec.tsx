import * as React from "react";
import { create } from "react-test-renderer";

import Spinner from "./Spinner";

describe("<Spinner />", () => {
    it("should render", async () => {
        const component = create(<Spinner isLoading={true} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("should not render when loading is set to false", async () => {
        const component = create(<Spinner isLoading={false} />);
        const instance = component.root;

        expect(instance.props.isLoading).toBe(false);
        expect(component.toJSON()).toBeNull();
    });

    it("should render when loading is set to true", async () => {
        const component = create(<Spinner isLoading={true} />);
        const instance = component.root;

        expect(instance.props.isLoading).toBe(true);
        expect(component.toJSON().type).toBe('div');
    });
});
