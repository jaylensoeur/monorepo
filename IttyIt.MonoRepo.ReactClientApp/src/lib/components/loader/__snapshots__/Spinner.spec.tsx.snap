// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render 1`] = `
<div
  data-testid="spinner"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
